rules/rules-reviewed/eap8/eap6/eap8-xml-binding.windup.xml

```diff
@@ -0,0 +1,161 @@
+                </when>
+                <perform>
+                    <hint title="Marshaller property package names must be updated" effort="1" category-id="mandatory">
+                        <message>Marshaller property package names must be updated to Jakarta XML Binding 4.0 for EAP8.
+                            Change the property constant name from `com.sun.xml.bind.*` to `org.glassfish.jaxb`</message>
+                        <link title="Jakarta XML Binding 4.0 specification"
+                              href="https://jakarta.ee/specifications/xml-binding/4.0/"/>
+                        <tag>eap8</tag>
+                    </hint>
+                </perform>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="eap8-xml-binding-00003">
+            <when>
+                <filecontent filename="{*}.xjb" pattern="http://java.sun.com/xml/ns/jaxb" />
+            </when>
+            <perform>
+                <hint title="The URI in the JAXB binding file should be updated" effort="1" category-id="mandatory">
+                    <message>The URI in JAXB binding files should be updated to Jakarta XML Binding 4.0 for EAP8.
+                        The uri must change from `http://java.sun.com/xml/ns/jaxb` to `https://jakarta.ee/xml/ns/jaxb`</message>
+                    <link title="Jakarta XML Binding 4.0 specification"
+                          href="https://jakarta.ee/specifications/xml-binding/4.0/"/>
+                    <tag>eap8</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="eap8-xml-binding-00004">
+            <when>
+                <or>
+                    <file filename="jaxb.properties"/>
+                    <file filename="javax.xml.bind.JAXBContext"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Deprecated steps in the implementation lookup algorithm have been removed" effort="1" category-id="mandatory">
+                    <message>Deprecated steps in the implementation lookup algorithm have been removed in Jakarta XML Binding 4.0 for EAP8.
+                        Use of the removed JAXBContext implementation discovery mechanisms should be replaced by one of the supported mechanisms discussed in the links.</message>
+                    <link title="Jakarta XML Binding 4.0 specification"
+                          href="https://jakarta.ee/specifications/xml-binding/4.0/"/>
+                    <link title="Implementation discovery in Jakarta XML Binding 4.0"
+                          href="https://jakarta.ee/specifications/xml-binding/4.0/jakarta-xml-binding-spec-4.0.html#implementation-discovery"/>
+                    <tag>eap8</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="eap8-xml-binding-00005">
+            <when>
+                <javaclass references="javax.xml.bind.JAXBContext.newInstance({*})" as="instanceMethods">
```

Comment:
Will this rule detect particular keys in the map passed?

Reply:
Not exactly no, it detects classes using these instantiation methods and searches within those classes for the offending class names. So it will detect the desired conditions but I imagine there will be some false positives on occasion.